client/app/reducers/requests.coffee

```diff
@@ -1,15 +1,93 @@
+# TODO:
+# This way to handle request is not very serious
+# we have to fin a more consistant solution
+# for temporary data persistence
+
 Immutable = require 'immutable'
 {ActionTypes, RequestStatus, Requests} = require '../constants/app_constants'
 
 DEFAULT_STATE = Immutable.Map
-    "#{Requests.DISCOVER_ACCOUNT}":     status: null, res: undefined
-    "#{Requests.CHECK_ACCOUNT}":        status: null, res: undefined
-    "#{Requests.ADD_ACCOUNT}":          status: null, res: undefined
-    "#{Requests.INDEX_MAILBOX}":        status: null, res: undefined
-    "#{Requests.REFRESH_MAILBOX}":      status: null, res: undefined
-    "#{Requests.FETCH_CONVERSATION}":   status: null, res: undefined
+    requests: Immutable.Map {inflight: null, queue: []}
+    error: Immutable.Map()
+    success: Immutable.Map()
 
 module.exports = (state = DEFAULT_STATE, action) ->
+
+    _addRequest = (name) ->
```

Comment:
@aenario : I changed the way to handle `Requests` because it was difficult to know in the old way to know `current` requests. Assuming that requests a executed synchronously we can use a queue that make all easier (to my mind).